storage/replica_command.go

```diff
@@ -2732,7 +2745,17 @@ func (r *Replica) ChangeReplicas(
 				},
 			},
 		})
-		return txn.Run(b)
+		if err := txn.Run(b); err != nil {
+			return err
+		}
+
+		if oldDesc.RangeID != 0 && !reflect.DeepEqual(oldDesc, desc) {
```

Comment:
Why are you doing this check here instead of above right after you read `oldDesc`?